File: code.java
//Пример бинарной кучи на java

import java.util.PriorityQueue;

public class Main {
    public static void main(String[] args) {
        PriorityQueue<Integer> heap = new PriorityQueue<>(); // min-куча
        heap.add(3);
        heap.add(1);
        heap.add(2);
        System.out.println(heap.poll());  // Выведет 1 (минимум)
    }
}

//Пример биноминальной кучи
class BinomialNode {
    int key;
    List<BinomialNode> children = new ArrayList<>();
    BinomialNode(int k) { key = k; }
}
BinomialNode n = new BinomialNode(5);
System.out.println(n.key); // 5

// Фибоначчиева куча
class FibNode {
    int key;
    List<FibNode> children = new ArrayList<>();
    FibNode(int k) { key = k; }
}
FibNode node = new FibNode(7);
System.out.println(node.key); // 7


//пример hash таблицы на java

import java.util.HashMap;

public class Main {
    public static void main(String[] args) {
        HashMap<String, Integer> map = new HashMap<>();
        map.put("one", 1);
        map.put("two", 2);
        System.out.println(map.get("one"));  // Выведет 1
    }
}

File: code.cc
//пример бинарной кучи на с++
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    vector<int> heap = {3, 1, 2};
    make_heap(heap.begin(), heap.end()); // Создание кучи
    pop_heap(heap.begin(), heap.end());  // Извлечение максимума
    heap.pop_back();
    cout << heap.front() << endl;  // Теперь максимальным элементом будет 2
    return 0;
}

//пример биноминальной кучи
struct BinomialNode {
    int key;
    std::vector<BinomialNode*> children;
    BinomialNode(int k) : key(k) {}
};
BinomialNode* n = new BinomialNode(5);

//Фибоначчиева куча
struct FibNode {
    int key;
    std::vector<FibNode*> children;
    FibNode(int k) : key(k) {}
};
FibNode* node = new FibNode(7);


//Пример hash таблицы
#include <iostream>
#include <unordered_map>
using namespace std;

int main() {
    unordered_map<string, int> map;
    map["one"] = 1;
    map["two"] = 2;
    cout << map["one"] << endl;  // Выведет 1
    return 0;
}
